sklearn/utils/tests/test_encode.py

```diff
@@ -53,21 +64,145 @@ def test_encode_with_check_unknown():
+    # check for check_unknown with missing values with object dtypes
+    values = np.array(['d', 'c', 'a', 'b', missing_value], dtype=object)
+    uniques = np.array(['c', 'a', 'b', missing_value], dtype=object)
+    if pickle_uniques:
+        uniques = pickle.loads(pickle.dumps(uniques))
 
-    assert_array_equal(diff, expected_diff)
-    assert_array_equal(valid_mask, expected_mask)
+    expected_diff = ['d']
+    expected_mask = [False, True, True, True, True]
+    _assert_check_unknown(values, uniques, expected_diff, expected_mask)
+
+    values = np.array(['d', 'c', 'a', 'b', missing_value], dtype=object)
+    uniques = np.array(['c', 'a', 'b'], dtype=object)
+    if pickle_uniques:
+        uniques = pickle.loads(pickle.dumps(uniques))
+
+    expected_diff = ['d', missing_value]
+
+    expected_mask = [False, True, True, True, False]
+    _assert_check_unknown(values, uniques, expected_diff, expected_mask)
+
+    values = np.array(['a', missing_value], dtype=object)
+    uniques = np.array(['a', 'b', 'z'], dtype=object)
+    if pickle_uniques:
+        uniques = pickle.loads(pickle.dumps(uniques))
+
+    expected_diff = [missing_value]
+    expected_mask = [True, False]
+    _assert_check_unknown(values, uniques, expected_diff, expected_mask)
+
+
+@pytest.mark.parametrize('missing_value', [np.nan, None, float('nan')])
+@pytest.mark.parametrize('pickle_uniques', [True, False])
+def test_unique_util_missing_values_objects(missing_value, pickle_uniques):
+    # check for _unique and _encode with missing values with object dtypes
+    values = np.array(['a', 'c', 'c', missing_value, 'b'], dtype=object)
+    expected_uniques = np.array(['a', 'b', 'c', missing_value], dtype=object)
+
+    uniques = _unique(values)
+
+    if missing_value is None:
+        assert_array_equal(uniques, expected_uniques)
+    else:  # missing_value == np.nan
+        assert_array_equal(uniques[:-1], expected_uniques[:-1])
+        assert np.isnan(uniques[-1])
+
+    if pickle_uniques:
+        uniques = pickle.loads(pickle.dumps(uniques))
+
+    encoded = _encode(values, uniques=uniques)
+    assert_array_equal(encoded, np.array([0, 2, 2, 3, 1]))
+
+
+def test_unique_util_missing_values_numeric():
+    # Check missing values in numerical values
+    values = np.array([3, 1, np.nan, 5, 3, np.nan], dtype=float)
+    expected_uniques = np.array([1, 3, 5, np.nan], dtype=float)
+    expected_inverse = np.array([1, 0, 3, 2, 1, 3])
+
+    uniques = _unique(values)
+    assert_array_equal(uniques, expected_uniques)
+
+    uniques, inverse = _unique(values, return_inverse=True)
+    assert_array_equal(uniques, expected_uniques)
+    assert_array_equal(inverse, expected_inverse)
+
+    encoded = _encode(values, uniques=uniques)
+    assert_array_equal(encoded, expected_inverse)
+
+
+def test_unique_util_with_both_missing_values():
+    # test for both types of missing values for object dtype
+    values = np.array([np.nan, 'a', 'c', 'c', None, float('nan'),
```

Comment:
If we're supporting this, please test the case of only NaN in training, but None present in test, and vice-versa.

Reply:
Added test to `test_one_hot_encoder_specified_categories`